android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/DataStoreMultiInsertTestKotlin.kt

```diff
@@ -0,0 +1,1954 @@
+        var result: Int? = null
+        var error: Throwable? = null
+
+        override fun onSuccess(result: Int?) {
+            this.result = result
+            finish()
+        }
+
+        override fun onFailure(error: Throwable) {
+            this.error = error
+            finish()
+        }
+
+        internal fun finish() {
+            latch.countDown()
+        }
+    }
+
+    private class DefaultKinveyReadCallback<T : GenericJson> (private val latch: CountDownLatch) : KinveyReadCallback<T> {
+        var result: KinveyReadResponse<T>? = null
+        var error: Throwable? = null
+
+        override fun onSuccess(result: KinveyReadResponse<T>) {
+            this.result = result
+            finish()
+        }
+
+        override fun onFailure(error: Throwable) {
+            this.error = error
+            finish()
+        }
+
+        internal fun finish() {
+            latch.countDown()
+        }
+    }
+
+    private class DefaultKinveyPushCallback (private val latch: CountDownLatch) : KinveyPushCallback {
+        var result: KinveyPushResponse? = null
+        var error: Throwable? = null
+
+        override fun onSuccess(result: KinveyPushResponse) {
+            this.result = result
+            finish()
+        }
+
+        override fun onFailure(error: Throwable) {
+            this.error = error
+            finish()
+        }
+
+        override fun onProgress(current: Long, all: Long) {
+
+        }
+
+        internal fun finish() {
+            latch.countDown()
+        }
+    }
+
+    private class DefaultKinveySyncCallback (private val latch: CountDownLatch) : KinveySyncCallback {
+        var kinveyPushResponse: KinveyPushResponse? = null
+        var kinveyPullResponse: KinveyPullResponse? = null
+        var error: Throwable? = null
+
+        override fun onSuccess(kinveyPushResponse: KinveyPushResponse, kinveyPullResponse: KinveyPullResponse) {
+            this.kinveyPushResponse = kinveyPushResponse
+            this.kinveyPullResponse = kinveyPullResponse
+            finish()
+        }
+
+        override fun onPullStarted() {
+
+        }
+
+        override fun onPushStarted() {
+
+        }
+
+        override fun onPullSuccess(kinveyPullResponse: KinveyPullResponse) {
+            this.kinveyPullResponse = kinveyPullResponse
+        }
+
+        override fun onPushSuccess(kinveyPushResponse: KinveyPushResponse) {
+            this.kinveyPushResponse = kinveyPushResponse
+        }
+
+        override fun onFailure(error: Throwable) {
+            this.error = error
+            finish()
+        }
+
+        internal fun finish() {
+            latch.countDown()
+        }
+    }
+
+    private fun createPerson(name: String): Person {
+        return Person(name)
+    }
+
+    @Throws(InterruptedException::class)
+    private fun <T : GenericJson> delete(store: DataStore<T>, query: Query): DefaultKinveyDeleteCallback {
```

Comment:
maybe better to move those methods like `delete`, `find`, `sync`, etc to a base class so it can be used by other tests in the future

Reply:
Moved common code and these methods to base class `BaseDataStoreTest`.